noinst/specs/specs_from_pdf.R

```diff
@@ -0,0 +1,356 @@
+  unlist() |>
+  trimws() |>
+  tibble() |>
+  setNames("line") |>
+  # Pull lines that are part of the desired tables
+  mutate(
+    starts_table = re_matches(line, rex("name", anything, "description of table")),
+    is_after_table = re_matches(line, rex("Name gives the name of each")),
+    table_number = cumsum(starts_table)
+  ) |>
+  group_by(table_number) |>
+  mutate(
+    spec = re_matches(
+      line,
+      rex(
+        # n.b. "Argument(s) for" is not sufficient here as there are also tables
+        #      "XXX Available Using the YYY Argument for SPEC"
+        #      those may reference the tables we want but actually document
+        #      different parameters of the spec
+        or(
+          "Output Tables for",
+          "Tables Saved As Percentages in the save Argument",
+          "Available Output Tables in Both print and save Arguments",
+          "Output Tables Available Only with save Argument"
+        ),
+        anything,
+        space,
+        capture(anything, name = "X"),
+        end
+      ),
+      options = "i"
+    )$X) |>
+  mutate(spec = trimws(tolower(spec))) |>
+  fill(spec, .direction = "downup") |>
+  filter(cumsum(is_after_table) == 0) |>
+  ungroup() |>
+  filter(!is.na(spec)) |> # Remove tables whose footer did not match that of desired tables (e.g. 7.46)
+  filter(table_number > 0) |>
+  filter(!starts_table) |>
+  # Mark lines that are just the continuation of the description col of the previous
+  # entry (they should contain only single spaces as they do not span multiple columns)
+  mutate(
+    is_part_of_prev = !re_matches(line, rex(between(spaces, 2, 5)))
+  ) |>
+  mutate(
+    entry_number = cumsum(!is_part_of_prev)
+  ) |>
+  summarize(
+    # gsub to remove no longer needed hyphenation
+    line = gsub("- ", "", paste(line, collapse = " ")),
+    .by = c("entry_number", "spec")
+  ) |>
+  mutate(
+    data = re_matches(
+      line,
+      rex(
+        start,
+        capture(alnums, name = "long"),
+        spaces,
+        capture(alnums, name = "short"),
+        capture(any_of(space, "·", "+"), name = "issave"),
+        capture(anything, name = "description"),
+        end
+      )
+    )
+  ) |>
+  unnest(data) |>
+  transmute(
+    long = sprintf("%s.%s", spec, long),
+    short,
+    is.save = re_matches(issave, "\\+"),
+    spec,
+    description
+  )
+
+jnd <- full_join(quick_from_full, full, by = "long")
+
+jnd |> filter(is.na(short.x))
+
+jnd |> filter(is.na(short.y))
+
+jnd |> filter(!is.na(short.x) & !is.na(short.y)) |>
+  filter(is.save.x != is.save.y)
+
+tbl_all <- full_join(quick_from_full, full, by = c("long")) |>
+  mutate(
+    short = coalesce(short.x, short.y),
+    spec = coalesce(spec.x, spec.y),
+    # Be optimistic for starters
+    is.save = is.save.x | is.save.y
+  ) |>
+  select(
+    -short.x,
+    -short.y,
+    -spec.x,
+    -spec.y,
+    -is.save.x,
+    -is.save.y
+  )
+
+manual_specs <- read_csv("noinst/specs/SPECS_MANUAL.csv")
```

Comment:
Could you add a bit of context here (or in a `README.md`). What is the purpose of `SPECS_MANUAL.csv`? When should it be updated? etc.

Reply:
Will do.
I'm still not quite clear on the meaning of `is.series`. Could we not just say everything that has `is.save == TRUE` is attempted to be read into `model$series`? If so all that really remains to be added manually is `requires`.

Reply:
I am sure there was a reason for `is.series`, so I would not drop it. Analyzing the existing entries that are `is.save` but not `is.series` would probably clarify why they are there. Let's try to keep as close to the existing version as possible.